### DISCUSSION TITLE
1.8.0

### DISCUSSION THREAD

**Discussion by gavlyukovskiy** [gavlyukovskiy/spring-boot-data-source-decorator#69] [General]
### ⭐ New Features
 - Delegation of instrumentation to Spring Cloud Sleuth when both are on classpath
 - #65 Add support for @DataJpaTest
### 🐞 Bug fixes
 - #61 DataSource decorator with HikariDataSourcePoolMetadata
 - #64 BeanDefinitionOverrideException using sleuth-3.1.0-RC1

<hr /><em>This discussion was created from the release <a href='https://github.com/gavlyukovskiy/spring-boot-data-source-decorator/releases/tag/1.8.0'>1.8.0</a>.</em>

**Comment by Taylor-Woods** [comment]
when I use  "public class DynamicDataSource extends AbstractRoutingDataSource",DynamicDataSource  can't Injection ,but version1.7.1 can success

**Reply by gavlyukovskiy** [reply]
Hi, that's caused by changing the proxying logic, which means you can not directly inject decorated datasource anymore (that was possible, but never documented as supported). I suggest you to inject a DataSource instead.
